### PR TITLE
[Test] Add DeepSeekV3.2 NSA Indexer Test Suite

### PR DESCRIPTION
**Pull request by Johnsonms**
## Motivation
Address the issue https://github.com/sgl-project/sglang/issues/12509

## Modifications
Added 11 tests:
  1. test_forward_decode_mode
  2. test_forward_extend_mode
  3. test_indexer_basic_creation
  4. test_indexer_metadata_interface
  5. test_indexer_with_alt_stream
  6. test_indexer_with_different_topk
  7. test_indexer_with_fused_wk
  8. test_rotate_activation
  9. test_rotate_activation_invalid_size
  10. test_shape_sanity_checks

## Benchmarking and Profiling
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/test/attention/test_nsa_indexer.py [hunk 1]
```diff
@@ -0,0 +1,639 @@
...
+
+        # Mean should be close to 0, variance close to 1
+        # Using lenient tolerance due to bfloat16 precision and quantization effects
+        mean_close = torch.allclose(mean, torch.zeros_like(mean), atol=1e-3)
+        var_close = torch.allclose(var, torch.ones_like(var), atol=5e-2)  # More lenient for variance
+
+        if not mean_close:
+            print(f"Mean check failed: max_abs_mean={mean.abs().max().item():.6f}")
+        if not var_close:
+            print(f"Var check failed: max_var_error={(var - 1.0).abs().max().item():.6f}")
```
[the remark this answers is not in the document]
Comment (Johnsonms) [reply python/sglang/test/attention/test_nsa_indexer.py]: Fixed

File: python/sglang/test/attention/test_nsa_indexer.py [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+from sglang.srt.layers import dp_attention as _dp_attn
+
+# Patch DP-attention globals before importing backends
+_dp_attn.get_attention_tp_size = lambda: 1  # TP size = 1 for unit test
+
+from sglang.srt.configs.model_config import AttentionArch
+from sglang.srt.layers.attention.nsa.nsa_indexer import (
+    BaseIndexerMetadata,
+    Indexer,
+    V32LayerNorm,
```
Comment (Fridge003) [inline python/sglang/test/attention/test_nsa_indexer.py]: Can you rebase the latest main and try again? 
V32LayerNorm has been deprecated in https://github.com/sgl-project/sglang/pull/12044

File: test/srt/layers/attention/nsa/test_nsa_indexer.py [hunk 3]
```diff
@@ -0,0 +1,628 @@
+import unittest
```
Comment (Fridge003) [inline test/srt/layers/attention/nsa/test_nsa_indexer.py:1]: Can we add this test to nightly CI?
It can be put under `test/srt`

Comment (Johnsonms) [reply test/srt/layers/attention/nsa/test_nsa_indexer.py:1]: Changed

File: python/sglang/test/attention/test_nsa_indexer.py (hunk 2 above)
Comment (Johnsonms) [reply python/sglang/test/attention/test_nsa_indexer.py]: 1. Rebased.
2. Removed V32LayerNorm tests since LayerNorm kernel now includes its own testing.

File: test/srt/layers/attention/nsa/test_nsa_indexer.py (hunk 3 above)
Comment (Johnsonms) [reply test/srt/layers/attention/nsa/test_nsa_indexer.py:1]: 1. Moved test_nsa_indexer.py to test/srt/layers/attention/nsa/test_nsa_indexer.py.
2. It will now run as part of the nightly-1-gpu test suite.

**Review by Fridge003 — APPROVED** [review]
